### ISSUE TITLE
Replace legacy code in `TextCNN` example 

### ISSUE THREAD

**Issue by KickItLikeShika** [pytorch/ignite#1900]
## 🚀 Feature
`torchtext` in their latest release `0.9.0` moved a lot of features to legacy code, and because of that we temporary updated our [TextCNN](https://github.com/pytorch/ignite/blob/master/examples/notebooks/TextCNN.ipynb) example to compatible with the legacy code in #1890 (as quick fix for now).
Now we need to replace the legacy code in this example with the provided features in `torchtext==0.9.0`.
- `torchtext==0.9.0` release note: https://github.com/pytorch/text/releases/tag/v0.9.0-rc5
- Migration guide from `torchtext` to show how to replace legacy code: https://github.com/pytorch/text/blob/master/examples/legacy_tutorial/migration_tutorial.ipynb

**Comment by cozek** [comment]
I did try to migrate it has have not been successful so far and haven't had time to go debug. Here is a link to the notebook
https://colab.research.google.com/drive/1LVyYWtSzY2Y65AO95gkG3Uo8rkiTRtA0?usp=sharing

I got stuck with the error:
```
ERROR:ignite.engine.engine.Engine:Engine run is terminating due to exception: Accuracy must have at least one example before it can be computed.
ERROR:ignite.engine.engine.Engine:Engine run is terminating due to exception: Accuracy must have at least one example before it can be computed.
```

IMO, perhaps we can keep the legacy notebook and add a fresh new one since it would require rewrite of the tutorial for the parts which used the legacy code.

**Comment by KickItLikeShika** [comment]
Thanks for the idea @cozek. @vfdev-5 what do you think about @cozek idea? he can send a PR to edit the name of existing `TextCNN` one to (for example `TextCNNLegacy`) and make a new notebook compatible with the new release of `torchtext`

**Comment by vfdev-5** [comment]
@cozek thanks for investigations. I think there is no value to keep legacy notebook as API changed. 
In our notebook we can add a requirement to install newer torchtext package. Yes, I agree that it would require rewriting a bit the tutorial as well.

@KickItLikeShika could you please check the error in the colab that has @cozek ?
It should be certainly related to the input dataloaders...

**Comment by KickItLikeShika** [comment]
Thanks for your nice work @cozek! as @vfdev-5 said the issue is related to the dataloaders, i have edited the way we create the dataloader (if you want to keep the bucket iterators you have to edit the `batch_sampler`), please check this copy notebook: https://colab.research.google.com/drive/16_F6G_cBmp9RJRXwhRbof36pcWd7EZ44?usp=sharing .
This notebook solves the `ERROR:ignite.engine.engine.Engine:Engine run is terminating due to exception: Accuracy must have at least one example before it can be computed.` issue, but we need to edit the devices related code in the model (and other training and evaluation functions if needed) before sending a PR, because you will run into issues like `Inputs and outputs are not on the same device` while running this notebook on GPU.

**Comment by cozek** [comment]
@KickItLikeShika Thanks for your help. Please grant me access so that I can view the NB. :) I think I can fix the device related issues.

**Comment by KickItLikeShika** [comment]
@cozek you can access the NB now, Good luck!
